06week/higherOrder.js

```diff
@@ -2,102 +2,145 @@
 
 const assert = require('assert');
 
+
 function forEach(arr, callback) {
-  // Your code here
+    // new array with converted items
+    let result = [];
+
+    // loop through entire length of array
+    for (let i = 0; i < arr.length; i++) {
+        result.push(callback(arr[i]));
```

Comment:
in forEach, you do not return the results of applying the callback function to the elements, you just execute the callback function on every element.